lib_i2c/src/i2c_pullup_common.xc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2025 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include <i2c.h>
+#include <xs1.h>
+#include <xclib.h>
+#include <timer.h>
+
+/** Check if pull-up resistors are present on SCL and SDA lines for two-port I2C.
+ *  This function drives the lines low briefly, then releases them
+ *  and checks if they return to high state within a reasonable time.
+ *  
+ *  \param p_scl       SCL port
+ *  \param p_sda       SDA port
+ *  
+ *  \returns I2C_ACK if both pull-ups are present,
```

Comment:
I2C_ACK used to signify something else - an I2S ACK. Its confusing to reuse it for something else here.